lib/auth/native/native.go

```diff
@@ -264,6 +264,9 @@ func (k *Keygen) GenerateUserCertWithoutValidation(c services.UserCertParams) ([
 	}
 	if c.ClientIP != "" {
 		cert.Permissions.Extensions[teleport.CertExtensionClientIP] = c.ClientIP
+		if c.PinSourceIP {
+			cert.Permissions.CriticalOptions = map[string]string{"source-address": c.ClientIP}
```

Comment:
I think we should add a bit more context of what `source-address` means/enforces
Maybe add a link to the OpenSSH docs? https://cvsweb.openbsd.org/src/usr.bin/ssh/PROTOCOL.certkeys?annotate=HEAD

I also noted that the docs says it requires CIDR notation
Would it work with just the IP? or would we need to add `/32` to turn this into a valid CIDR notation?
The golang's ssh server implementation supports both: https://github.com/golang/crypto/blob/2cf3adece1227c48e1673f1c37d70357e1a6b9d3/ssh/server.go#L306-L319
I'm not sure about the OpenSSH implementation https://github.com/openssh/openssh-portable/blob/2dc328023f60212cd29504fc05d849133ae47355/addrmatch.c#L109

Reply:
I've created const here and added a comment, hopefully it's clearer now.
Good catch with CIDR, I've added `/32` so we are always safe.